tests/test_generated.py

```diff
@@ -0,0 +1,81 @@
+from google.api_core.exceptions import BadRequest, NotFound
+from google.cloud import bigquery
+from .util import coerce_result, generate_tests
+
+import json
+import os
+import pytest
+
+
+@pytest.fixture(scope="session")
+def bq():
+    return bigquery.Client(project=os.environ.get("GCP_PROJECT", None))
+
+
+@pytest.fixture(params=list(generate_tests()))
+def generated_test(request):
+    return request.param
+
+
+@pytest.fixture
+def dataset(bq, generated_test):
+    # create dataset
+    dataset_id = generated_test.get_dataset_id()
+    try:
+        bq.get_dataset(dataset_id)
+    except NotFound:
+        bq.create_dataset(dataset_id)
+    # wait for test
+    yield bq.dataset(dataset_id)
+    # clean up
+    try:
+        bq.delete_dataset(dataset_id).result()
+    except BadRequest as e:
+        if str(e).endswith("still in use."):
+            pass  # ignore still in use exception on delete_dataset
```

Comment:
iiuc this exception indicates that there are still tables in this dataset, so either `tables` needs to clean up, or there might be a setting to recursively delete

Reply:
Recursively deleting a dataset is something the Java helpers handle, so I'm guessing not built straight into the API.

Reply:
I just needed to set `delete_contents=True`